lib/logstash/filters/elasticsearch/client.rb

```diff
@@ -32,6 +36,20 @@ def search(params)
         @client.search(params)
       end
 
+      def self.validate_config!(user, password, options={})
+        ssl     = options[:ssl]
+        hosts   = options[:hosts]
+        logger  = options[:logger]
+        unless ssl.nil?
+          if hosts.detect {|h| h =~ %r(^https?://) }
```

Comment:
a couple things here:
 - mixing `unless` and `if` in nested blocks is a bit confusing to me
 - while `ssl` will be nil at this point if `options` didn't include the `:ssl` directive, that is more of an _affect_ and may be clearer to test `options` for `:ssl` directly.
 - `String#start_with?(*other)` is more readable than regex (and faster to execute, too).

consider:

~~~ ruby
        if options.include?(:ssl) && hosts.any? { |h| h.start_with?('http://', 'https://') }
          logger.error # ...
~~~